homeassistant/components/intellifire/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for IntelliFire integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from intellifire4py import IntellifireAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+async def validate_input(hass: HomeAssistant, host: str) -> str:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    api = IntellifireAsync(host)
+    try:
+        await api.poll()
+    except ConnectionError as exception:
+        raise CannotConnect from exception
+
+    # Return the serial number which will be used to calculate a unique ID for the device/sensors
+    return api.data.serial
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for IntelliFire."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+        errors = {}
+
+        try:
+            serial = await validate_input(self.hass, user_input[CONF_HOST])
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+            LOGGER.exception("Connection Exception")
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception("Unexpected Exception")
+            errors["base"] = "unknown"
+        else:
+            await self.async_set_unique_id(serial)
+            self._abort_if_unique_id_configured(
+                updates={CONF_HOST: user_input[CONF_HOST]}
+            )
+
+            return self.async_create_entry(
+                title="Fireplace",
```

Comment:
Is it common for users to have multiple instances / fireplaces? Wouldn't it make sense to list the IP address in the name as well? So that not all entries are called 'Fireplace'.

Reply:
I think that case is rare... even if one has: They can rename it if they like.